focus/operations/src/status.rs

```diff
@@ -2,15 +2,52 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use anyhow::Result;
-use focus_internals::model::repo::Repo;
+use focus_internals::{model::repo::Repo, target::TargetTypes};
 use focus_util::app::{App, ExitCode};
-use std::{path::Path, sync::Arc};
+use std::{collections::HashSet, path::Path, sync::Arc};
 
-pub fn run(sparse_repo: impl AsRef<Path>, app: Arc<App>) -> Result<ExitCode> {
+pub fn run(
+    sparse_repo: impl AsRef<Path>,
+    app: Arc<App>,
+    targets_flag: bool,
+    target_types: Vec<TargetTypes>,
+) -> Result<ExitCode> {
+    let target_types = HashSet::<TargetTypes>::from_iter(target_types.iter().cloned());
     let repo = Repo::open(sparse_repo.as_ref(), app)?;
     let selections = repo.selection_manager()?;
     let selection = selections.selection()?;
-    println!("{}", selection);
+    if target_types.is_empty() && !targets_flag {
+        println!("{}", selection);
+    } else {
+        let mut targets = selection.targets;
+        if !targets_flag {
```

Comment:
One side effect of doing it this way is that when you use `--targets` you will also get mandatory targets/projects in your output since you're using `compute_complete_target_set`, which returns ALL the targets from the selection and mandatory set. Is that what you intended in this case?